CenterForOpenScience/ember-osf-web
# [EMB-369][Embosf] Settings page and connected emails

## Purpose

Adds the settings page and the connected emails component.

## Summary of Changes

- Add settings page
- Create connected-emails component
- Update mirage
- Add tests

<img width="848" alt="screen shot 2018-10-30 at 10 49 22 am" src="https://user-images.githubusercontent.com/19379783/47727138-3cb88000-dc32-11e8-9957-d5d30e1b5e75.png">

<img width="623" alt="screen shot 2018-10-30 at 10 52 30 am" src="https://user-images.githubusercontent.com/19379783/47727143-40e49d80-dc32-11e8-8a0a-4a307e0a0c8e.png">

<img width="625" alt="screen shot 2018-10-30 at 10 52 57 am" src="https://user-images.githubusercontent.com/19379783/47727155-4641e800-dc32-11e8-93aa-11750950a5bf.png">

<img width="832" alt="screen shot 2018-10-30 at 10 54 16 am" src="https://user-images.githubusercontent.com/19379783/47727162-49d56f00-dc32-11e8-9e5c-213db75afd81.png">


## Side Effects

N/A

## Feature Flags

N/A

## QA Notes

This will use a few new updates:
- Loading indicators will show up when the user is first loading the page
- Emails are now displayed in a paginated list
- Any time a user updates an email (ex. make it primary), the email list will show a skeleton screen
- A modal will pop up whenever a user deletes or adds an email

Please be sure to test that all of the original functionality and these new changes work as expected.

## Ticket


https://openscience.atlassian.net/browse/EMB-369

## Notes

- Will need to be updated after 'resend confirmation email' functionality is added to the backend.

This PR will need a few different things to work:
- https://github.com/CenterForOpenScience/ember-osf-web/pull/435 (for the updates to mirage)
- https://github.com/CenterForOpenScience/ember-osf-web/pull/452 (for the validated-model-form)
- Work will also need to be done on the paginated list

# Reviewer Checklist
